anchor-service/src/anchor_batch.rs

```diff
@@ -14,31 +21,121 @@ use crate::{
 /// ceramic_anchor_service::Store is responsible for fetching AnchorRequests and storing TimeEvents.
 #[async_trait]
 pub trait Store: Send + Sync {
-    /// Get a batch of AnchorRequests.
-    async fn local_sourced_data_events(&self) -> Result<Vec<AnchorRequest>>;
     /// Store a batch of TimeEvents.
-    async fn put_time_events(&self, batch: TimeEventBatch) -> Result<()>;
+    async fn insert_many(&self, items: Vec<TimeEventInsertable>, informant: NodeId) -> Result<()>;
+    /// Get a batch of AnchorRequests.
+    async fn events_since_high_water_mark(
+        &self,
+        informant: NodeId,
+        high_water_mark: i64,
+        limit: i64,
+    ) -> Result<Vec<AnchorRequest>>;
 }
 
 /// An AnchorService is responsible for anchoring batches of AnchorRequests and storing TimeEvents generated based on
 /// the requests and the anchor proof.
 pub struct AnchorService {
     tx_manager: Arc<dyn TransactionManager>,
-    _event_service: Arc<dyn Store>,
-    _batch_linger_time: Duration,
+    event_service: Arc<dyn Store>,
+    high_water_mark_store: HighWaterMarkStore,
+    node_id: NodeId,
+    anchor_interval: Duration,
+    anchor_batch_size: u64,
 }
 
 impl AnchorService {
     /// Create a new AnchorService.
     pub fn new(
         tx_manager: Arc<dyn TransactionManager>,
-        _event_service: Arc<dyn Store>,
-        _batch_linger_time: Duration,
+        event_service: Arc<dyn Store>,
+        pool: SqlitePool,
+        node_id: NodeId,
+        anchor_interval: Duration,
+        anchor_batch_size: u64,
     ) -> Self {
         Self {
             tx_manager,
-            _event_service,
-            _batch_linger_time,
+            event_service,
+            high_water_mark_store: HighWaterMarkStore::new(pool),
+            node_id,
+            anchor_interval,
+            anchor_batch_size,
+        }
+    }
+
+    /// Run the AnchorService:
+    /// - Get anchor requests from the AnchorClient
+    /// - Deduplicate the anchor requests
+    /// - Anchor the batch using a Transaction Manager
+    /// - Store the TimeEvents using the AnchorClient
+    ///
+    /// This function will run indefinitely, or until the process is shutdown.
+    pub async fn run(&mut self, shutdown_signal: impl Future<Output = ()>) {
```

Comment:
I do not see any tests of this method. Can we an an integration test that shows this basic flow working?